### PR TITLE
Update MoveToMapPokemon to use events instead of logger.

### PR DESCRIPTION
**Pull request by chrisle**
Short Description: 

Update `MoveToMapPokemon` to use events instead of logger. Includes documentation and a refactor of the module.

Fixes:
- Deprecation warnings emitted by MoveToMapPokemon.

### CODE REVIEW

**Comment by eriweb** [comment]
No longer getting warnings, LGTM

**Comment by douglascamata** [comment]
👍 

@TheSavior, please review this when you have time.

File: pokemongo_bot/cell_workers/move_to_map_pokemon.py [hunk 1]
```diff
@@ -182,18 +254,111 @@ def work(self):
...
+            data=self._pokemon_event_data(pokemon)
+        )
+
+    def _teleport_back(self):
+        """Teleports trainer back to their last position."""        
+        last_position = self.bot.position[0:2]
+        self.emit_event(
+            'move_to_map_pokemon_teleport_back',
+            formatted=('Teleporting back to previous location ({last_lat}, '
+                       '{last_long})'),
```
Comment (chypriote) [inline pokemongo_bot/cell_workers/move_to_map_pokemon.py]: Should be last_lon

File: pokemongo_bot/__init__.py [hunk 2]
```diff
@@ -393,6 +393,35 @@ def _register_events(self):
...
+            'move_to_map_pokemon_fail',
+            parameters=('message',)
+        )
+        self.event_manager.register_event(
+            'move_to_map_pokemon_updated_map',
+            parameters=('lat', 'lon')
+        )
+        self.event_manager.register_event(
+            'move_to_map_pokemon_teleport_to',
+            parameters=('poke_name', 'poke_dist', 'poke_lat', 'poke_lon',
```
Comment (TheSavior) [inline pokemongo_bot/__init__.py]: It think the shorthand typically used for longitude is `lng`.

**Comment by TheSavior** [comment]
This seems fine.

**Comment by TheSavior** [comment]
:+1:

**Comment by mhdasding** [comment]
could we remove the `move_to_map_pokemon_` prefix for the events? they're already namespaced by the task emitting the events and no other task is prefexing its events.

**Comment by solderzzc** [comment]
👍

**Comment by Jamezin** [comment]
Im getting this error from using this. It shows this right when encountering pokemon (When sniping) and before trying to teleport back to location.

`Traceback (most recent call last):
  File "pokecli.py", line 503, in <module>
    main()
  File "pokecli.py", line 89, in main
    bot.tick()
  File "C:\Users\Alika\Desktop\PokemonGo-Bot-dev\pokemongo_bot\__init__.py", line 434, in tick
    if worker.work() == WorkerResult.RUNNING:
  File "C:\Users\Alika\Desktop\PokemonGo-Bot-dev\pokemongo_bot\cell_workers\move_to_map_pokemon.py", line 255, in work
    return self.snipe(pokemon)
  File "C:\Users\Alika\Desktop\PokemonGo-Bot-dev\pokemongo_bot\cell_workers\move_to_map_pokemon.py", line 214, in snipe
    self._teleport_back()
  File "C:\Users\Alika\Desktop\PokemonGo-Bot-dev\pokemongo_bot\cell_workers\move_to_map_pokemon.py", line 339, in _teleport_back
    data={'last_lat': last_position[0], 'last_lon': last_position[1]}
  File "C:\Users\Alika\Desktop\PokemonGo-Bot-dev\pokemongo_bot\base_task.py", line 27, in emit_event
    data=data
  File "C:\Users\Alika\Desktop\PokemonGo-Bot-dev\pokemongo_bot\event_manager.py", line 61, in emit
    formatted_msg = formatted.format(**data)
KeyError: 'last_long'`

**Pull request merged**